state/futures/selectors.ts

```diff
@@ -878,50 +878,109 @@ export const selectSlTpModalInputs = createSelector(
+					? unserialized.notionalValue.div(unserialized.margin).abs()
+					: wei(0),
+			};
+		}
+		return null;
 	}
 );
 
-export const selectEditPosDesiredFillPrice = createSelector(
+export const selectEditPositionPreview = createSelector(
+	selectFuturesType,
 	selectEditPositionInputs,
-	selectMarketPrice,
-	({ nativeSizeDelta }, marketPrice) => {
-		const priceImpact = getDefaultPriceImpact('market');
-		const impactDecimalPercent = priceImpact.div(100);
-		return Number(nativeSizeDelta) < 0
-			? marketPrice.mul(wei(1).sub(impactDecimalPercent))
-			: marketPrice.mul(impactDecimalPercent.add(1));
+	(state: RootState) => state.futures,
+	(type, { nativeSizeDelta }, futures) => {
+		const preview = futures[accountType(type)].previews.edit;
+		const unserialized = preview ? unserializePotentialTrade(preview) : null;
+		if (unserialized) {
+			const priceImpact = getDefaultPriceImpact('market');
+			const impactDecimalPercent = priceImpact.div(100);
+			const desiredFillPrice =
+				Number(nativeSizeDelta) < 0
+					? unserialized.price.mul(wei(1).sub(impactDecimalPercent))
+					: unserialized.price.mul(impactDecimalPercent.add(1));
+			return {
+				...unserialized,
+				desiredFillPrice,
+				leverage: unserialized.margin.gt(0)
+					? unserialized.notionalValue.div(unserialized.margin).abs()
+					: wei(0),
+			};
+		}
+		return null;
 	}
 );
 
-export const selectClosePosDesiredFillPrice = createSelector(
+export const selectClosePositionPreview = createSelector(
+	selectFuturesType,
 	selectEditPositionModalInfo,
 	selectClosePositionOrderInputs,
-	({ position, marketPrice }, { price, orderType }) => {
-		const priceImpact = getDefaultPriceImpact(orderType);
-
-		const impactDecimalPercent = priceImpact.div(100);
-		let orderPrice = orderType === 'market' ? marketPrice : wei(price?.value || 0);
-		orderPrice = orderPrice ?? wei(0);
-		return position?.position?.side === PositionSide.LONG
-			? orderPrice.mul(wei(1).sub(impactDecimalPercent))
-			: orderPrice.mul(impactDecimalPercent.add(1));
+	(state: RootState) => state.futures,
+	(type, { position }, { price, orderType }, futures) => {
+		const preview = futures[accountType(type)].previews.close;
+		const unserialized = preview ? unserializePotentialTrade(preview) : null;
+		if (unserialized) {
+			const priceImpact = getDefaultPriceImpact(orderType);
+			const impactDecimalPercent = priceImpact.div(100);
+			let orderPrice =
+				(orderType === 'market' ? unserialized.price : wei(price?.value || 0)) ?? wei(0);
+			const desiredFillPrice =
+				position?.position?.side === PositionSide.LONG
+					? orderPrice.mul(wei(1).sub(impactDecimalPercent))
+					: orderPrice.mul(impactDecimalPercent.add(1));
+			return {
+				...unserialized,
+				desiredFillPrice,
+				leverage: unserialized.margin.gt(0)
+					? unserialized.notionalValue.div(unserialized.margin).abs()
+					: wei(0),
+			};
+		}
+		return null;
 	}
 );
 
+export const selectDesiredTradeFillPrice = createSelector(
```

Comment:
I think we can delete all these desiredFillPrice selectors now and just read the value straight from preview selectors